config/routes.rb

```diff
@@ -0,0 +1,22 @@
+Rails.application.routes.draw do
+
+  root 'main#index'
+
+
+  get 'login', to: 'users#login_form', as: 'login'
+  post 'login', to: 'users#login'
+  get 'logout', to: 'users#logout', as: 'logout'
+
+  get 'topten', to: 'works#topten', as: 'topten'
+
+  resources :users
+
+  resources :works
+    resources :votes, only: [:new, :create]
```

Comment:
I'm not sure if this is what you intended, but this syntax will not nest your vote routes under works. To do so, you need to add a `do`/`end`:

```ruby
resources :works do
  resources :votes, only: [:new, :create]
end
```